documents/wg-application.md

```diff
@@ -0,0 +1,93 @@
+This should lead to experimenting with different approaches on how to use it and creating awesome tooling.
+
+## Is your WG long-running or temporary?
+
+In our current vision, the WG should live until we fulfill our goals.
+
+In the end, we hope the WG will evolve into another one to cover similar topics:
+to support other GPGPU platforms or to create higher-level frameworks to improve end-to-end experience based on community feedback.
+
+## What is your long-term vision?
+
+Having a reliable and safe CUDA development experience is our ultimate goal.
+This should include:
+
+* Getting `nvptx64-nvidia-cuda` to a [Tier 2](https://forge.rust-lang.org/platform-support.html) support state.
+* Test infrastructure on a real hardware and running the tests during Rust CI process.
+* Rich documentation, references, tutorials and examples.
+* A broad set of new compile errors, warnings and lints for SIMT execution model to avoid pitfalls and ensure code soundness.
+
+## How do you expect the relationship to the project be?
+
+The WG will be responsible for CUDA related issues and user requests.
+This includes [already reported issues](https://github.com/rust-lang/rust/issues?q=is%3Aopen+is%3Aissue+label%3AO-NVPTX) and those that will be opened due to the work made by the WG.
+An important aspect is that the WG will take care of the state of the `rust-ptx-linker` and will do its best to avoid blocking someone else's work (like it, unfortunately, happened in [rust#59752](https://github.com/rust-lang/rust/pull/59752)).
+
+### How do you want to establish accountability?
+
+For that purpose, we will publish an agenda and made decisions from our meetings.
+Once we achieve important milestones we plan to make public announces.
+
+## Which other WGs do you expect to have close contact with?
+
+We would like to cooperate with Core or Compiler Teams in discussions about safety in SIMT code.
+Additionally, it would be very important to discuss a strategy of reliable deployment of `rust-ptx-linker` (likely as a `rustup` component).
+
+On the other hand, proposed [Machine Learning WG](https://internals.rust-lang.org/t/enabling-the-formation-of-new-working-groups/10218/11) could leverage CUDA accelerated computing capabilities, so we can discuss their use cases.
+
+## What are your short-term goals?
+
+Mainly, short-term goals are about evaluating and discussing how can we achieve long-term goals:
+
+* Make a poll about community experiences and use cases for CUDA.
+* Deploy the `rust-ptx-linker` as a `rustup` component.
+* Collect soundness and safety issues that can happen in SIMT execution model.
+* Decide testing approach on a real hardware.
+
+## Who is the initial leadership?
+
+> TBD...
+
+## How do you intend to make your work accessible to outsiders?
+
+Excessive learning materials and retrospective about made decisions should help to get more people involved in either discussions or further experimenting.
```

Comment:
I think this would be a good place to mention that some members write blog posts. "Excessive learning materials" sound excessive. I mean, our initial roadmap is just about getting the target to work reliably. Ideally, compiling a CUDA kernel should just work out-of-the box. If we have to teach anything here to get that to compile, we kind of failed. I expect most of the documentation to be in the "frameworks" that allow you to combine CUDA kernels with normal Rust code, launch kernels, etc. But that would be something that we could tackle after the target works reliably.

Reply:
That makes sense, thanks